### PR TITLE
Expose wasm_runtime_get_exec_env_singleton to the API users

### PR DESCRIPTION
**Pull request by yamt**
Sometimes it can be useful to access the singleton.
Eg. use wasm_runtime_set_user_data on it.

### CODE REVIEW

File: core/iwasm/common/wasm_runtime_common.h [hunk 1]
```diff
@@ -592,9 +592,6 @@ wasm_runtime_call_indirect(WASMExecEnv *exec_env, uint32 element_indices,
 bool
 wasm_runtime_create_exec_env_singleton(WASMModuleInstanceCommon *module_inst);
 
-WASMExecEnv *
```
Comment (wenyongh) [inline core/iwasm/common/wasm_runtime_common.h:595]: Had better not remove the declaration here, just add WASM_RUNTIME_API_EXTERN like others:
```C
WASM_RUNTIME_API_EXTERN WASMExecEnv *
wasm_runtime_get_exec_env_singleton(WASMModuleInstanceCommon *module_inst);
```

Comment (yamt) [reply core/iwasm/common/wasm_runtime_common.h:595]: i moved this to wasm_export.h. do you mean to have it in both headers?

Comment (yamt) [reply core/iwasm/common/wasm_runtime_common.h:595]: i think i got it now. although i feel it's a weird convention, i updated the patch to follow it. thank you.

Comment (wenyongh) [reply core/iwasm/common/wasm_runtime_common.h:595]: Yes, prefer to have it in both files.

**Pull request merged**